table/block_based/block_based_table_reader.cc

```diff
@@ -1219,9 +1225,7 @@ Status BlockBasedTable::GetDataBlockFromCache(
   assert(block->IsEmpty());
   const Cache::Priority priority =
       rep_->table_options.cache_index_and_filter_blocks_with_high_priority &&
-              (block_type == BlockType::kFilter ||
-               block_type == BlockType::kCompressionDictionary ||
-               block_type == BlockType::kIndex)
+              block_type != BlockType::kData
```

Comment:
Won't this set the priority for other blocks type as well like `kRangeDeletion`, `kMetaIndex` as `High` or is this intentional?

Reply:
Yes, good noticing and this was mostly intentional. Right now those other kinds of metadata blocks are not inserted into block cache, but I believe we intend for "index_and_filter" options to generally apply to all metadata blocks involved in the read path. "index_and_filter" is just legacy naming that was not forward-looking.

Forward-looking: one exception I can think of is that we might start putting table properties into block cache, and since those don't service normal reads, they do not (IMHO) deserve the high-priority treatment. So perhaps I'll add `block_type != BlockType::kProperties` for such a future. Thoughts @ajkr @hx235 ?

I'm also considering that this is a performance-critical code path, so efficiency matters here. (The compiler might not do a great job with many `||` `==` even if it's possible to avoid branching.)

Reply:
> one exception I can think of is that we might start putting table properties into block cache, and since those don't service normal reads, they do not (IMHO) deserve the high-priority treatment. So perhaps I'll add && block_type != BlockType::kProperties

I don't see a problem of doing this. 

> I'm also considering that this is a performance-critical code path, so efficiency matters here. (The compiler might not do a great job with many || == even if it's possible to avoid branching.)

Would a static local const set here help reduce branching?